sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_arg.asciidoc

```diff
@@ -386,6 +511,271 @@ constant property.
+and `annotated_arg<T2, ...>`. Let `operatorOP` denote the operator used.
+The overloaded operator `operatorOP` utilizes `operatorOP(T2)` of type `T`
+and is available only if `operatorOP(T2)` is valid for objects of type `T`.
+The value and result is the same as the result of `operator+(T2)` applied to the
+objects of type `T` and `T2`.
+
+[frame="topbot",options="header"]
+|===
+|Overloaded Operators
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() + std::declval<T2>())>
+R operator+(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() - std::declval<T2>())>
+R operator-(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2
+          typename R = decltype(std::declval<T>() * std::declval<T2>())>
+R operator*(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() / std::declval<T2>())>
+R operator/(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() % std::declval<T2>())>
+R operator%(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() & std::declval<T2>())>
+R operator&(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() \| std::declval<T2>())>
+R operator\|(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() ^ std::declval<T2>())>
+R operator^(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() >> std::declval<T2>())>
+R operator>>(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() << std::declval<T2>())>
+R operator<<(const annotated_arg<T2, PropertyList2> &other) const;
+----
+
+|===
+
+The following table describes the overloaded operators for types `annotated_arg<T, ...>`
```

Comment:
@gmlueck Also here, I have made the table such that the operator definition with annotated_arg on LHS is on one column and the definition with annotated_arg on RHS is on the other column. Let me know if this looks ok.